### PR TITLE
Miscellany part deux

### LINKED ISSUE #73: Update data guidelines.
# Test notes
* The Data Guidelines page in the wiki details all required and options renderer settings that serve to map input data.

### LINKED ISSUE #100: Adding additional filters changes existing controls
# Test Notes
- Add the filters below and verify that the other controls do not change from multi-selects to drop downs

Add this in CAT settings to add two filters:
{ "filters": [ { "value_col": "querytext", "label": "Text" }, { "value_col": "queryresolveddt", "label": "ResolveDate" } ] }

and notice how all the other controls go from multi-selects to drop downs.

### LINKED ISSUE #102: Truncate free text variables to 100 characters...
...and display the remaining text on hover.

**Test notes**
Switch Maximum Length of Cell Text from 100 to 5. Check the cells to make sure they're truncating.

### LINKED ISSUE #104: Change Group by dropdown to radio buttons.
_Status Group_, not _Group by_

### LINKED ISSUE #115: Update settings schema for v2.0.0.
# Test notes
* Settings form appears as expected in CAT.

### PR DESCRIPTION
**Pull request by samussiah**
# Issues
Closes #73 
Closes #100 
Closes #102 
Closes #104 
Closes #115

### CODE REVIEW

File: build/queryOverview.js [hunk 1]
```diff
@@ -1107,6 +1105,64 @@
             });
     }
 
+    function addGroupByHighlight() {
```
Comment (pburnsdata) [inline build/queryOverview.js:1119]: this is a nice touch

File: scripts/check-settings-schema.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+//renderer settings
+const settings = require('../src/configuration/rendererSettings.js').default();
+const actualSettings = Object.keys(settings);
+
+//differences
+const missingExpectedSettings = actualSettings
+    .filter(setting => expectedSettings.indexOf(setting) < 0);
+if (missingExpectedSettings.length > 0) {
+    console.log('\x1b[31m%s\x1b[0m', 'These settings are missing from the settings schema:\n');
```
Comment (pburnsdata) [inline scripts/check-settings-schema.js:18]: good idea to check if schema has all the keys

File: scripts/build-data-guidelines-wiki.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            markdown.push(
+                `|**${
+                    variable.setting}[]**|_user-defined_|${
+                    variable.type}|${
+                    variable.description}|${
+                    variable.required ? '**Y**' : ''
+                }|`
+            );
+    } else
+        console.warn(`This wiki can't handle ${variable.type}s! Get outta here!`);
```
Comment (pburnsdata) [inline scripts/build-data-guidelines-wiki.js:66]: Yeah! Get outta here!

File: src/chart/onResize/addNoDataIndicator.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    if (this.filtered_data.length === 0)
+        this.svg
+            .append('text')
+            .classed('qo-no-data', true)
+            .attr({
+                x: this.plot_width / 2,
+                y: this.plot_height / 2,
+                'text-anchor': 'middle'
+            })
+            .text('No queries selected.  Verify that no filter selections are in conflict.');
```
Comment (pburnsdata) [inline src/chart/onResize/addNoDataIndicator.js:13]: Well said, well said

**Review by pburnsdata — APPROVED** [review]
Looks good! Like the red text on hover and the schema jazz

**Comment by danedexF5** [comment]
Passed feature testing

**Pull request merged**